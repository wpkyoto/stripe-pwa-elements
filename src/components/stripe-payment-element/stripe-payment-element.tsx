import { Component, Host, h, Element } from '@stencil/core';
import { Prop } from '@stencil/core/internal';
import { loadStripe } from "@stripe/stripe-js";
import { waitForElm } from '../../utils/utils';

@Component({
  tag: 'stripe-payment-element',
  shadow: false,
})
export class StripePaymentElement {
  @Element() el: HTMLStripePaymentElement;
  /**
   * Your Stripe publishable API key.
   */
  @Prop() publishableKey: string;

  constructor() {
    if (this.publishableKey) {
        loadStripe(this.publishableKey)
          .then(async stripe => {
            if (!stripe) return
            const elements = stripe.elements({
              mode: 'payment',
              amount: 1000,
              currency: 'jpy'
            })
            const paymentElement = elements.create('payment')
            const target = await waitForElm(this.el, '#payment-element')
            paymentElement.mount(target)
          })
    }
  }

  render() {
    return (
      <Host>
        <div id="payment-element"></div>
      </Host>
    );
  }

}
