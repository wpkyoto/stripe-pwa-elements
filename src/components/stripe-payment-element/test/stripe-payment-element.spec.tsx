import { newSpecPage } from '@stencil/core/testing';
import { StripePaymentElement } from '../stripe-payment-element';

describe('stripe-payment-element', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [StripePaymentElement],
      html: `<stripe-payment-element></stripe-payment-element>`,
    });
    expect(page.root).toMatchSnapshot()
  });
});
