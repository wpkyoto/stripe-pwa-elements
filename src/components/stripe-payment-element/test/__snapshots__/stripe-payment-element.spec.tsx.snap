// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stripe-payment-element renders 1`] = `
<stripe-payment-element>
  <div id="payment-element"></div>
</stripe-payment-element>
`;
